src/app/session.cpp

```diff
@@ -317,7 +317,7 @@ void Session::Forge::iconize(QString name, QAbstractButton *button, uint8_t size
 void Session::Forge::set_icon(QString name, QAbstractButton *button, bool dynamic) const
 {
     button->setIcon(QIcon(QString(":/icons/%1.svg").arg(name)));
-    button->setProperty("colorized",false);
+    button->setProperty("colorized", true);
```

Comment:
This is not correct. The value should be `false` The logic is that icon buttons always have a property `colorized` that indicates whether or not it has been colorized. It is set to `false` for all buttons until it is colorized, at which point it is set to `true`.

Reply:
Also note that setting this to `true` by default causes slowness and excessive resource usage, for example when switching from light ↔ dark mode.

Reply:
Ok, then I guess problem lies in variable `dynamic`default value `false`:
It should be true, or we need to update every place where iconized buttons are used (many places).

https://github.com/openDsh/dash/blob/b985a43a34a20dade264e323578a4d55a4ec7674/include/app/session.hpp#L143-L144

Reply:
The dynamic coloring of a button is only possible if the button has been "colorized" and AFAIK in every case the button is colorized, the `colorized` variable is set to `true`

https://github.com/openDsh/dash/blob/93af01fd02f18de520259b787e11d0aebe09da94/src/app/session.cpp#L128

Reply:
If you want a button to be dynamically colored then you need to call `set_icon()` with `dynamic=true` for that button. For the buttons I think you are talking about, I don't think they're supposed to be colorized but if that's the intent then that change needs to be made.

Reply:
Problem is that in dark theme those buttons are black berelly visible...

Reply:
so yeah the icons have gotten a bit bloated with trying to handle everything... comments could definitely have helped clear things up

but the way things should be working is an icon could have a couple of states (outside of the standard qt icon states)
- base color
- accent color
- alternate icon

for iconize, the dynamic param is used to color an icon that can't be done lazily (ex at runtime)

anytime the session needs to be updated, every icon was being colorized because the colors need to be updated depending on light/dark mode and theme

we only really need to color the icons that were visible on the UI, but it was just easier at the time to recolor all icons

now I'm working on having the icons be painted with the correct colors to make all this recoloring moot

Reply:
All right, then I believe we can close this PR and wait for better solution.
At least this conversation was valuable to clear things :)
Thx!